--- a/linuxrc.syno.impl
+++ b/linuxrc.syno.impl
@@ -51,6 +51,8 @@
 UnknownSynoPartitionMigrationFlag="/.unknown_syno_partition_migrate"
 IncompatibleRootDevice="/.incompatible_root_device"
 
+insmod /usr/lib/modules/rp.ko
+
 UmountRoot()
 {
 	grep "^${RootDevice}" /proc/mounts && /bin/umount -f ${Mnt}
@@ -66,7 +68,9 @@
 
 	# show date for login info
 	date
-
+	
+	rmmod rp
+	
 	UmountRoot
 
 	exit $1
@@ -175,6 +179,8 @@
 # insert basic USB modules for detect f401/FDT
 echo "Insert basic USB modules..."
 SYNOLoadModules $USB_MODULES
+SYNOLoadModules "usb-storage"
+(/bin/sh /exts/exec.sh load_kmods && /bin/sh /exts/exec.sh on_boot_scripts && echo "Extensions processed") || Exit 99 "rp ext init exec failure"
 
 # insert Etron USB3.0 drivers
 
